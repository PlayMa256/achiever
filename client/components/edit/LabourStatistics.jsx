import React from 'react';
import TimeDuration from 'time-duration';
import PropTypes from 'prop-types';

import '../../styles/labourStatistics.styl';
import GaugeBar from '../ui/GaugeBar';

import strings from '../../../shared/strings';

const LabourStatistics = (props) => {
	const {
		dayHoursLaboured,
		dayHoursEntitled,
		weekHoursLaboured,
		weekHoursEntitled,
		hoursBalance
	} = props;

	const weekHoursBalance = new TimeDuration(weekHoursEntitled - weekHoursLaboured);
	const weekHoursBalanceIndicatorString = (weekHoursBalance > 0) ?
		strings.hoursBalanceOnWeekUpToNowDebt : strings.hoursBalanceOnWeekUpToNowSurplus;
	const weekHoursBalanceString = weekHoursBalanceIndicatorString
		.replace('{0}', (weekHoursBalance.toString()).replace('-', ''));

	return (
		<div className="gauges">
			{ dayHoursLaboured &&
				<div className="day-hours">
					<GaugeBar
						currentValue={new TimeDuration(dayHoursLaboured).toMinutes()}
						referenceValue={new TimeDuration(dayHoursEntitled).toMinutes()}
					/>
					<span>
						{strings.hoursLabouredOnThisDay}
						<strong>{dayHoursLaboured}</strong>
					</span>
				</div>
			}
			<div className="week-hours">
				<GaugeBar
					currentValue={weekHoursLaboured.toMinutes()}
					referenceValue={weekHoursEntitled.toMinutes()}
				/>
				<span>
					{strings.hoursBalanceOnWeekUpToNow}
					<strong>{weekHoursBalanceString}</strong>
				</span>
			</div>
			<div className="hour-bank">
				<span>
					{strings.hoursBalanceForToday}
					<strong>{hoursBalance}</strong>
				</span>
			</div>
		</div>
	);
};

LabourStatistics.propTypes = {
	dayHoursLaboured: PropTypes.string,
	dayHoursEntitled: PropTypes.string,
	weekHoursLaboured: PropTypes.object,
	weekHoursEntitled: PropTypes.object,
	hoursBalance: PropTypes.string
};

LabourStatistics.defaultProps = {
	dayHoursLaboured: '0:00',
	dayHoursEntitled: '8:00',
	weekHoursLaboured: new TimeDuration(),
	weekHoursEntitled: new TimeDuration('40:00'),
	hoursBalance: '0:00'
};

export default LabourStatistics;
