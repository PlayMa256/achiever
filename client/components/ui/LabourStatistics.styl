@import "../../styles/_colours.styl"
@import "../../styles/_standards.styl"

.LabourStatistics
    box-style();
    padding: 9px 10px 6px;
    width: 245px;
    text-align: right;
    margin: 10px 0;
    overflow: hidden;
