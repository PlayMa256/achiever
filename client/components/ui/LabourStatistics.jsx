import React from 'react';
import PropTypes from 'prop-types';

import Gauge from './Gauge';
import GaugeDescription from './GaugeDescription';

import strings from '../../../shared/strings';
import './LabourStatistics.styl';

const LabourStatistics = ({
	dayBalance,
	weekBalance,
	contractedTime,
	weekDay,
	totalBalance
}) => (
	<div className="LabourStatistics">
		<Gauge
			description={strings.hoursLabouredOnThisDay}
			value={dayBalance}
			expected={contractedTime}
		/>
		<Gauge
			description={strings.hoursBalanceOnWeekUpToNow}
			value={weekBalance}
			expected={contractedTime * weekDay}
		/>
		<GaugeDescription
			description={strings.hoursBalanceForToday}
			value={totalBalance}
		/>
	</div>
);

LabourStatistics.propTypes = {
	totalBalance: PropTypes.number,
	contractedTime: PropTypes.number,
	dayBalance: PropTypes.number,
	weekBalance: PropTypes.number,
	weekDay: PropTypes.number.isRequired
};

LabourStatistics.defaultProps = {
	totalBalance: 0,
	contractedTime: 8,
	dayBalance: 0,
	weekBalance: 0
};

export default LabourStatistics;
