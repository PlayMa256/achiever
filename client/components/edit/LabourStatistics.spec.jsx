import React from 'react';
import { shallow } from 'enzyme';
import TimeDuration from 'time-duration';

import LabourStatistics from './LabourStatistics';

const defaultProps = {
	dayHoursEntitled: '8:00',
	dayHoursLaboured: '8:00',
	rawBalance: '2:20',
	weekHoursEntitled: new TimeDuration(960),
	weekHoursLaboured: new TimeDuration(840)
};

describe('LabourStatistics', () => {
	describe('render', () => {
		it('should properly render with surplus or debit', () => {
			const wrapperSurplus = shallow(<LabourStatistics />);
			const wrapperDebt = shallow(<LabourStatistics />);

			wrapperDebt.setProps(defaultProps);
			wrapperSurplus.setProps({ ...defaultProps, weekHoursLaboured: new TimeDuration(1000) });
			expect(wrapperDebt).toMatchSnapshot();
			expect(wrapperSurplus).toMatchSnapshot();
		});
	});
});
