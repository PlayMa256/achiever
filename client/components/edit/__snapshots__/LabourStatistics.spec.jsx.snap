// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabourStatistics render should properly render with surplus or debit 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabourStatistics
    dayHoursEntitled="8:00"
    dayHoursLaboured="8:00"
    rawBalance="2:20"
    weekHoursEntitled={
      TimeDuration {
        "_minutes": 960,
      }
    }
    weekHoursLaboured={
      TimeDuration {
        "_minutes": 840,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="day-hours"
        >
          <GaugeBar
            currentValue={480}
            referenceValue={480}
            verticalBar={false}
          />
          <span>
            Horas trabalhadas no dia:
            <strong>
              8:00
            </strong>
          </span>
        </div>,
        <div
          className="week-hours"
        >
          <GaugeBar
            currentValue={840}
            referenceValue={960}
            verticalBar={false}
          />
          <span>
            Na semana até esse dia
            <strong>
              devendo 2:00
            </strong>
          </span>
        </div>,
        <div
          className="hour-bank"
        >
          <span>
            Balanço de horas atual:
            <strong>
              0:20
            </strong>
          </span>
        </div>,
      ],
      "className": "gauges",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <GaugeBar
              currentValue={480}
              referenceValue={480}
              verticalBar={false}
            />,
            <span>
              Horas trabalhadas no dia:
              <strong>
                8:00
              </strong>
            </span>,
          ],
          "className": "day-hours",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentValue": 480,
              "referenceValue": 480,
              "verticalBar": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Horas trabalhadas no dia:",
                <strong>
                  8:00
                </strong>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Horas trabalhadas no dia:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "8:00",
                },
                "ref": null,
                "rendered": "8:00",
                "type": "strong",
              },
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <GaugeBar
              currentValue={840}
              referenceValue={960}
              verticalBar={false}
            />,
            <span>
              Na semana até esse dia
              <strong>
                devendo 2:00
              </strong>
            </span>,
          ],
          "className": "week-hours",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentValue": 840,
              "referenceValue": 960,
              "verticalBar": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Na semana até esse dia",
                <strong>
                  devendo 2:00
                </strong>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Na semana até esse dia",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "devendo 2:00",
                },
                "ref": null,
                "rendered": "devendo 2:00",
                "type": "strong",
              },
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Balanço de horas atual:
            <strong>
              0:20
            </strong>
          </span>,
          "className": "hour-bank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Balanço de horas atual:",
              <strong>
                0:20
              </strong>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Balanço de horas atual:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "0:20",
              },
              "ref": null,
              "rendered": "0:20",
              "type": "strong",
            },
          ],
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="day-hours"
          >
            <GaugeBar
              currentValue={480}
              referenceValue={480}
              verticalBar={false}
            />
            <span>
              Horas trabalhadas no dia:
              <strong>
                8:00
              </strong>
            </span>
          </div>,
          <div
            className="week-hours"
          >
            <GaugeBar
              currentValue={840}
              referenceValue={960}
              verticalBar={false}
            />
            <span>
              Na semana até esse dia
              <strong>
                devendo 2:00
              </strong>
            </span>
          </div>,
          <div
            className="hour-bank"
          >
            <span>
              Balanço de horas atual:
              <strong>
                0:20
              </strong>
            </span>
          </div>,
        ],
        "className": "gauges",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <GaugeBar
                currentValue={480}
                referenceValue={480}
                verticalBar={false}
              />,
              <span>
                Horas trabalhadas no dia:
                <strong>
                  8:00
                </strong>
              </span>,
            ],
            "className": "day-hours",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currentValue": 480,
                "referenceValue": 480,
                "verticalBar": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Horas trabalhadas no dia:",
                  <strong>
                    8:00
                  </strong>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Horas trabalhadas no dia:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "8:00",
                  },
                  "ref": null,
                  "rendered": "8:00",
                  "type": "strong",
                },
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <GaugeBar
                currentValue={840}
                referenceValue={960}
                verticalBar={false}
              />,
              <span>
                Na semana até esse dia
                <strong>
                  devendo 2:00
                </strong>
              </span>,
            ],
            "className": "week-hours",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currentValue": 840,
                "referenceValue": 960,
                "verticalBar": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Na semana até esse dia",
                  <strong>
                    devendo 2:00
                  </strong>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Na semana até esse dia",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "devendo 2:00",
                  },
                  "ref": null,
                  "rendered": "devendo 2:00",
                  "type": "strong",
                },
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Balanço de horas atual:
              <strong>
                0:20
              </strong>
            </span>,
            "className": "hour-bank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Balanço de horas atual:",
                <strong>
                  0:20
                </strong>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Balanço de horas atual:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0:20",
                },
                "ref": null,
                "rendered": "0:20",
                "type": "strong",
              },
            ],
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LabourStatistics render should properly render with surplus or debit 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabourStatistics
    dayHoursEntitled="8:00"
    dayHoursLaboured="8:00"
    rawBalance="2:20"
    weekHoursEntitled={
      TimeDuration {
        "_minutes": 960,
      }
    }
    weekHoursLaboured={
      TimeDuration {
        "_minutes": 1000,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="day-hours"
        >
          <GaugeBar
            currentValue={480}
            referenceValue={480}
            verticalBar={false}
          />
          <span>
            Horas trabalhadas no dia:
            <strong>
              8:00
            </strong>
          </span>
        </div>,
        <div
          className="week-hours"
        >
          <GaugeBar
            currentValue={1000}
            referenceValue={960}
            verticalBar={false}
          />
          <span>
            Na semana até esse dia
            <strong>
              0:40 a mais
            </strong>
          </span>
        </div>,
        <div
          className="hour-bank"
        >
          <span>
            Balanço de horas atual:
            <strong>
              3:00
            </strong>
          </span>
        </div>,
      ],
      "className": "gauges",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <GaugeBar
              currentValue={480}
              referenceValue={480}
              verticalBar={false}
            />,
            <span>
              Horas trabalhadas no dia:
              <strong>
                8:00
              </strong>
            </span>,
          ],
          "className": "day-hours",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentValue": 480,
              "referenceValue": 480,
              "verticalBar": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Horas trabalhadas no dia:",
                <strong>
                  8:00
                </strong>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Horas trabalhadas no dia:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "8:00",
                },
                "ref": null,
                "rendered": "8:00",
                "type": "strong",
              },
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <GaugeBar
              currentValue={1000}
              referenceValue={960}
              verticalBar={false}
            />,
            <span>
              Na semana até esse dia
              <strong>
                0:40 a mais
              </strong>
            </span>,
          ],
          "className": "week-hours",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentValue": 1000,
              "referenceValue": 960,
              "verticalBar": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Na semana até esse dia",
                <strong>
                  0:40 a mais
                </strong>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Na semana até esse dia",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0:40 a mais",
                },
                "ref": null,
                "rendered": "0:40 a mais",
                "type": "strong",
              },
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Balanço de horas atual:
            <strong>
              3:00
            </strong>
          </span>,
          "className": "hour-bank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Balanço de horas atual:",
              <strong>
                3:00
              </strong>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Balanço de horas atual:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3:00",
              },
              "ref": null,
              "rendered": "3:00",
              "type": "strong",
            },
          ],
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="day-hours"
          >
            <GaugeBar
              currentValue={480}
              referenceValue={480}
              verticalBar={false}
            />
            <span>
              Horas trabalhadas no dia:
              <strong>
                8:00
              </strong>
            </span>
          </div>,
          <div
            className="week-hours"
          >
            <GaugeBar
              currentValue={1000}
              referenceValue={960}
              verticalBar={false}
            />
            <span>
              Na semana até esse dia
              <strong>
                0:40 a mais
              </strong>
            </span>
          </div>,
          <div
            className="hour-bank"
          >
            <span>
              Balanço de horas atual:
              <strong>
                3:00
              </strong>
            </span>
          </div>,
        ],
        "className": "gauges",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <GaugeBar
                currentValue={480}
                referenceValue={480}
                verticalBar={false}
              />,
              <span>
                Horas trabalhadas no dia:
                <strong>
                  8:00
                </strong>
              </span>,
            ],
            "className": "day-hours",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currentValue": 480,
                "referenceValue": 480,
                "verticalBar": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Horas trabalhadas no dia:",
                  <strong>
                    8:00
                  </strong>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Horas trabalhadas no dia:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "8:00",
                  },
                  "ref": null,
                  "rendered": "8:00",
                  "type": "strong",
                },
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <GaugeBar
                currentValue={1000}
                referenceValue={960}
                verticalBar={false}
              />,
              <span>
                Na semana até esse dia
                <strong>
                  0:40 a mais
                </strong>
              </span>,
            ],
            "className": "week-hours",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currentValue": 1000,
                "referenceValue": 960,
                "verticalBar": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Na semana até esse dia",
                  <strong>
                    0:40 a mais
                  </strong>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Na semana até esse dia",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "0:40 a mais",
                  },
                  "ref": null,
                  "rendered": "0:40 a mais",
                  "type": "strong",
                },
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Balanço de horas atual:
              <strong>
                3:00
              </strong>
            </span>,
            "className": "hour-bank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Balanço de horas atual:",
                <strong>
                  3:00
                </strong>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Balanço de horas atual:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "3:00",
                },
                "ref": null,
                "rendered": "3:00",
                "type": "strong",
              },
            ],
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
